mitiq/mitiq_qiskit/tests/test_zne_mitiq_qiskit.py

```diff
@@ -151,3 +167,43 @@ def test_mitigate_executor_with_shot_list():
                 "scale_factor": scale_factors[i],
                 "shots": shot_list[i],
             }
+
+
+@pytest.mark.parametrize("order", [(0, 1), (1, 0), (0, 1, 2), (1, 2, 0)])
+def test_measurement_order_is_preserved_single_register(order):
+    """Tests measurement order is preserved when folding, i.e., the dictionary
+    of counts is the same as the original circuit on a noiseless simulator.
+    """
+    qreg, creg = QuantumRegister(len(order)), ClassicalRegister(len(order))
+    circuit = QuantumCircuit(qreg, creg)
+
+    circuit.x(qreg[0])
+    for i in order:
+        circuit.measure(qreg[i], creg[i])
+
+    folded = zne.scaling.fold_gates_at_random(circuit, scale_factor=1.0)
+
+    assert get_counts(folded) == get_counts(circuit)
```

Comment:
Since the output state is |11...>, the results will be deterministic and always equal to 11...  . 
In this special case, does `assert get_counts(folded) == get_counts(circuit)` still detect a wrong measurement order?
Probably yes, but I am asking just to be sure.

Reply:
There's only one bit flip, so the output should always have Hamming weight one. Unless I'm misunderstanding you?

Reply:
In other words, you're right `assert get_counts(folded) == get_counts(circuit)` will not detect if the measurement order flipped if the state is |11...1>, but the state is never |11....1>.

Reply:
Sorry, I initially thought that all qubit were flipped. Now I see.